solutions/devsprint-cecilia-saraiva-2/FinanceAppTests/Modules/ActivityDetails/ActivityDetailsViewModelTests.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  ActivityDetailsViewModel.swift
+//  FinanceAppTests
+//
+//  Created by Leonardo Almeida on 22/08/22.
+//
+
+import Foundation
+import XCTest
+
+@testable import FinanceApp
+
+final class ActivityDetailsViewModelTests: XCTestCase {
+    
+    let delegate = ActivityDetailsViewModelDelegateStub()
+    
+    func test_WhenFecthData_ShouldReturnCountOne() {
+        // GIVEN
+        let financeServiceStub = FinanceServiceStub()
+        
+        var activityViewModelStub = ActivityDetailsViewModel(financeService: financeServiceStub)
+        activityViewModelStub.delegate = delegate
+        
+        // WHEN
+        activityViewModelStub.fetchData()
+        
+        let expectation = expectation(description: "test_WhenFecthData_ShouldReturnCountOne")
+        
+        // THEN
+        XCTAssertEqual(financeServiceStub.countFetchActivityDetails, 1)
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.5, execute: {
+            XCTAssertEqual(self.delegate.countDidFetchActivityDetails, 1)
+            
+            expectation.fulfill()
+        })
+        wait(for: [expectation], timeout: 2.0)
+    }
+}
+
+final class ActivityDetailsViewModelDelegateStub: ActivityDetailsViewModelDelegate {
+    var countDidFetchActivityDetails: Int = 0
+    
+    func didFetchActivityDetails(_ data: ActivityDetails) {
+        self.countDidFetchActivityDetails += 1
+    }
+}
+
+class FinanceServiceStub: FinanceServiceProtocol {
+    private(set) var countFetchActivityDetails: Int = 0
```

Comment:
[sugestao] similar a de acima:
```suggestion
    private(set) var fetchActivityDetailsCallCount: Int = 0
```